### PR TITLE
fix: fix issue where cdk bootstrap failed and add response streaming feature (toggle off). Close #122

### LINKED ISSUE #122: 'pnpm run galileo-cli deploy --dryRun' issue with SSO
### Describe the bug

this is the output of the command 'pnpm run galileo-cli deploy --dryRun':

[DRYRUN] docker info
✔ Config file name? … config.json
✔ Application Name (stack/resource naming) … Galileo
✔ AWS Profile … default
✔ AWS Region (app) … us-east-1
    CredentialsProviderError: Profile is configured with invalid SSO credentials. Required parameters "sso_account_id", "sso_region", "sso_role_name", "sso_start_url". Got sso_session, 
    sso_account_id, sso_role_name, region, output
    Reference: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html
 ELIFECYCLE  Command failed with exit code 1.

this is the content of my ~/.aws/config file:

[default]
sso_session = cloud9-sso
sso_account_id = XXX
sso_role_name = YYY
region = us-east-1
output = json
[sso-session cloud9-sso]
sso_start_url = https://ZZZ/start
sso_region = eu-central-1
sso_registration_scopes = sso:account:access

My SSO config works well for other application, so I do believe something is wrong with Galileo, could you pls. have a look ?

### Expected Behavior

'pnpm run galileo-cli deploy --dryRun' should work without any issue with SSO

### Current Behavior

this is the output of the command 'pnpm run galileo-cli deploy --dryRun':

[DRYRUN] docker info
✔ Config file name? … config.json
✔ Application Name (stack/resource naming) … Galileo
✔ AWS Profile … default
✔ AWS Region (app) … us-east-1
    CredentialsProviderError: Profile is configured with invalid SSO credentials. Required parameters "sso_account_id", "sso_region", "sso_role_name", "sso_start_url". Got sso_session, 
    sso_account_id, sso_role_name, region, output
    Reference: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html
 ELIFECYCLE  Command failed with exit code 1.

this is the content of my ~/.aws/config file:

[default]
sso_session = cloud9-sso
sso_account_id = XXX
sso_role_name = YYY
region = us-east-1
output = json
[sso-session cloud9-sso]
sso_start_url = https://ZZZ/start
sso_region = eu-central-1
sso_registration_scopes = sso:account:access

My SSO config works well for other application, so I do believe something is wrong with Galileo, could you pls. have a look ?

### Reproduction Steps

configure SSO and try to run 'pnpm run galileo-cli deploy --dryRun'

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### Environment details (OS name and version, etc.)

pnpm run prerequisite-check  > monorepo@0.0.0 prerequisite-check /home/ubuntu/environment/aws-genai-conversational-rag-reference > ./prerequisite-check.sh  Galileo Prerequisite Check...  system: Linux 6.5.0-1017-aws #17~22.04.2-Ubuntu SMP Mon Mar 25 20:28:54 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux pnpm: 9.0.5  node: v20.12.2  python: Python 3.11.0rc1  poetry: Poetry (version 1.8.2)  docker: Docker version 26.0.1, build d260a54  java: openjdk 18.0.2 2022-07-19 OpenJDK Runtime Environment Corretto-18.0.2.9.1 (build 18.0.2+9-FR) OpenJDK 64-Bit Server VM Corretto-18.0.2.9.1 (build 18.0.2+9-FR, mixed mode, sharing)  aws: aws-cli/2.15.38 Python/3.11.8 Linux/6.5.0-1017-aws exe/x86_64.ubuntu.22 prompt/off

### PR DESCRIPTION
**Pull request by jessieweiyi**
## Description
This PR includes @sperka PR #130. The PR has been tested on different use cases and fix a few issues:
* Revert the UI tsconfig files change to address the runtime UI issues
* ListChatMessages API failed due to validation when nextToken is null)

The PR also:
* Tested with langchain version 0.1.36 (Upgraded in PR #131) - The issue where only one chunk returned cannot be identified
* Allows users to config whether to stream response for Bedrock models

Issues found in the streaming feature, so add a feature flag on the feature to toggle it off:
* When issue occurs, the button is loading forever 
* The second requests do not get the update

To toggle on: add ?features=streaming query string to the url 

Pending To-do items:
* With streaming, it should implement a progress bar in chat panel
* upgrade `lambda-powertools` to v2

Here is a copied of original PR description: 

This PR introduces streaming support and was also targeting a bit of a cleanup/refresh.

* vscode settings update (use vscode v1.85+) - editor settings had breaking changes
* adjust projenrc setup to conform [aws-pdk](https://github.com/aws/aws-pdk) update breaking changes
* bump dependencies to `latest` or `latest-that-doesn't-break-anything`
* replace discontinued @aws-sdk packages with @smithy packages
* fix for #112 
* tsconfig settings update for apis and website
* fix first-deployment issue for admin user/group dependency
* add [Websocket API](https://aws.github.io/aws-pdk/developer_guides/type-safe-api/websocket_getting_started.html) from PDK's `type-safe-api`
* add streaming support to RAG chain and split `create-message` handler to support `REST` and `WS`
* add websocket connection management (via ddb persistance)
* restructure infra nested stack setup to avoid circular dependencies
* website updated to support websocket connections

There are a few things that should be added or changed (refer to `// TODO`s in the code):

1. Website 
    * make choice of streaming/rest call configurable
    * with streaming, it should implement a progress bar in chat panel
2. SDK (RAG chain)
   * Currently `langchain@0.0.194` version is used, there were huge improvements so updating to latest should be inevitable - however, this includes quite a bit of effort
   * Also, `lambda-powertools` v2 is out, but has breaking changes
   * `qaChain` definition and how streaming is handled, especially based on the models used. With the current generic approach, streaming is not happening, there is only one chunk returned (which is the actual result) instead of incremental answer (aka tokens). This _may be_ an issue because of the langchain version, _or_ something else. That part needs to be tested, both for bedrock LLMs and Sagemaker-hosted LLMs.
3. Websocket communication structure
    * Current implementation is not as restrictive as it should be, `operation` value can be any string value which should be restricted to `enum`, and maybe other fields could be limited as well -- needs thorough review 
4. Add docs

**Does this PR introduce a breaking change?**  
No breaking change. The streaming feature is under a feature toggle. 

## How Has This Been Tested?

* **What environment was this tested on?**

## PR Checklist

* [ ] Have you added/updated documentation?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran build and tests with your changes locally?

**IMPORTANT:** Please review the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: demo/website/src/hooks/ws-chats.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            messageId: nanoid(32),
+            text: input.payload as string,
+            type: 'human',
+          };
+          return draft;
+        });
+      }
+
+      if (input.operation === 'HandleSendMessage' && input.status === 'SUCCESS') {
+        // const result = input.payload as CreateChatMessageResponseContent;
```
Comment (martenpayne) [inline demo/website/src/hooks/ws-chats.ts]: dead code?

File: demo/infra/src/application/ai/inference-engine/engine/handler/ws-handler/index.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    messageId: tmpMessageId,
+    updatedAt: Date.now(),
+    operation: 'HandleSendMessage',
+    status: 'STARTING',
+    payload: input.question,
+  });
+
+  const result = await createMessage({
+    logger,
+
```
Comment (martenpayne) [inline demo/infra/src/application/ai/inference-engine/engine/handler/ws-handler/index.ts]: dead whitespace

File: demo/infra/src/application/presentation/index.ts [hunk 3]
```diff
@@ -193,16 +281,22 @@ export class PresentationStack extends NestedStack {
...
           actions: ['execute-api:Invoke'],
-          resources: [this.typesafeApi.api.arnForExecuteApi('*', '/*', '*')],
+          resources: [
+            this.typesafeApi.api.arnForExecuteApi('*', '/*', '*'),
+            this.wsApi.api.arnForExecuteApi('*', '/*', '*'),
+          ],
         }),
       ],
     });
+    // policy.attachToRole(props.authenticatedUserRole);
```
Comment (martenpayne) [inline demo/infra/src/application/presentation/index.ts]: dead code

File: packages/galileo-sdk/src/chat/callback.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  readonly updateStatus: UpdateStatusCallback;
+}
+
+export enum ChainOperation {
+  CLASSIFY = 'CLASSIFY',
+  CONDENSE_QUESTION = 'CONDENSE_QUESTION',
+  DOCUMENT_RETRIEVE = 'DOCUMENT_RETRIEVE',
+  QA = 'QA',
+}
+export interface ChainOperationPayload {
```
Comment (martenpayne) [inline packages/galileo-sdk/src/chat/callback.ts:25]: minor: add newline

File: packages/galileo-sdk/src/chat/engine.ts [hunk 5]
```diff
@@ -76,6 +81,9 @@ export class ChatEngine {
       retriever,
       verbose,
       returnTraceData,
+
```
Comment (martenpayne) [inline packages/galileo-sdk/src/chat/engine.ts]: extraneous newline

File: packages/galileo-sdk/src/chat/engine.ts [hunk 6]
```diff
@@ -100,6 +110,9 @@ export class ChatEngine {
       classifyChain,
       verbose,
       returnTraceData,
+
```
Comment (martenpayne) [inline packages/galileo-sdk/src/chat/engine.ts]: extraneous newline?

**Review by martenpayne — CHANGES_REQUESTED** [review]
couple minor comments

File: demo/website/src/components/chat/components/HumanInputForm.tsx [hunk 7]
```diff
@@ -23,8 +28,19 @@ export default function HumanInputForm(props: { chat: Chat; onSuccess?: () => vo
...
+    isBedrockFramework &&
+    useStreaming &&
+    (typeof options.llm?.useStreaming === 'undefined' || options.llm?.useStreaming);
+
   useEffect(() => {
-    if (!useStreaming) {
+    if (!isStreamingResponse) {
       setIsLoading(createChatMessage.isLoading);
     }
   }, [useStreaming, createChatMessage.isLoading]);
```
Comment (chn217) [inline demo/website/src/components/chat/components/HumanInputForm.tsx:30]: nit: the 2nd param of useEffect hook should include isStreamingResponse

**Review by chn217 — COMMENTED** [review]
Nice work on this PR! 👏

Just a general comment: I noticed that there are a few TODOs included in this PR. It would be good to track these TODOs in separate issues in the backlog to ensure they don't get lost or forgotten. 

The most important missing TODO seems to be related to documentation (DOC). We should create a separate issue for this and prioritize it accordingly.

**Review by martenpayne — APPROVED** [review]

**Pull request merged**